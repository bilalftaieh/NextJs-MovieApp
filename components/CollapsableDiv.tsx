'use client'
// Importing necessary libraries
import React, { useState } from "react";
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faArrowDown, faArrowRight } from "@fortawesome/free-solid-svg-icons";

// Props for CollapsableDiv component
interface CollapsableDivProps {
    name: string; // Name of the collapsable div
    children: React.ReactNode; // Children of the collapsable div
}

// The main function for the CollapsableDiv component
export default function CollapsableDiv({ name, children }: CollapsableDivProps) {
    // State for whether the div is expanded or not
    const [isExpanded, setIsExpanded] = useState(false);

    // Function to handle click
    const handleClick = () => {
        setIsExpanded(!isExpanded);
    }

    // Returning the JSX for the component
    return (
        <div className="flex flex-col bg-slate-700 text-gray-300 p-4 hover:bg-slate-800">
            <div className="collapsable-div flex flex-row items-center cursor-pointer p-2 
            rounded-md space-x-4" onClick={handleClick}>
                <h2 className="font-bold text-lg">{name}</h2>
                <div>
                    {!isExpanded && <FontAwesomeIcon icon={faArrowRight} />}
                    {isExpanded && <FontAwesomeIcon icon={faArrowDown} />}
                </div>
            </div>
            <div className="mt-2">
                {isExpanded && children}
            </div>
        </div>
    );
}
