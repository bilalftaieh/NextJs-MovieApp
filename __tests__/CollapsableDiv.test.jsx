import '@testing-library/jest-dom'
import { render, fireEvent, screen} from '@testing-library/react'
import CollapsableDiv from '@/components/CollapsableDiv'

it('correctly renders the Collapsable Div and expands on click', () => {
  const { container } = render(
    <CollapsableDiv name='language'>
      <h1>HELLO!</h1>
    </CollapsableDiv>
  );

  // Check initial state (not expanded)
  expect(container).toMatchSnapshot();

  // Simulate a click event
  fireEvent.click(screen.getByText(/language/i));

  // Check expanded state with a new snapshot
  expect(container).toMatchSnapshot();
});


