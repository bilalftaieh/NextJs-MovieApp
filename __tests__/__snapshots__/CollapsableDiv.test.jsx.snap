// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correctly renders the Collapsable Div and expands on click 1`] = `
<div>
  <div
    class="flex flex-col bg-slate-700 text-gray-300 p-4 hover:bg-slate-800"
  >
    <div
      class="collapsable-div flex flex-row items-center cursor-pointer p-2    rounded-md space-x-4"
    >
      <h2
        class="font-bold text-lg"
      >
        language
      </h2>
      <div>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-arrow-right "
          data-icon="arrow-right"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M438.6 278.6c12.5-12.5 12.5-32.8 0-45.3l-160-160c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L338.8 224 32 224c-17.7 0-32 14.3-32 32s14.3 32 32 32l306.7 0L233.4 393.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0l160-160z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <div
      class="mt-2"
    />
  </div>
</div>
`;

exports[`correctly renders the Collapsable Div and expands on click 2`] = `
<div>
  <div
    class="flex flex-col bg-slate-700 text-gray-300 p-4 hover:bg-slate-800"
  >
    <div
      class="collapsable-div flex flex-row items-center cursor-pointer p-2    rounded-md space-x-4"
    >
      <h2
        class="font-bold text-lg"
      >
        language
      </h2>
      <div>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-arrow-down "
          data-icon="arrow-down"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M169.4 470.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 370.8 224 64c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 306.7L54.6 265.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <div
      class="mt-2"
    >
      <h1>
        HELLO!
      </h1>
    </div>
  </div>
</div>
`;
